ui/vite.config.ts

```diff
@@ -1,14 +1,22 @@
-import { defineConfig } from "vite";
+import { defineConfig, loadEnv } from "vite";
 import react from "@vitejs/plugin-react";
 import tsconfigPaths from "vite-tsconfig-paths";
 
-export default defineConfig({
-  plugins: [tsconfigPaths(), react()],
-  server: {
-    port: 3000,
-  },
-  build: {
-    outDir: "./dist",
-    minify: "esbuild",
-  },
+export default defineConfig(({ mode }) => {
+  const env = loadEnv(mode, process.cwd(), "");
+  return {
+    plugins: [tsconfigPaths(), react()],
+    server: {
+      port: 3000,
+      proxy: {
+        "/api": {
+          target: env.VITE_PROXY_API_URL ?? "/",
+        },
+      },
```

Comment:
This change allows the requests to the API to proxied to an external API (e.g. for developing the UI externally from the API deployment).

This is required otherwise you get CORS errors when requesting from the API.